awsmp/models.py

```diff
@@ -139,3 +142,139 @@ def ami_access_role_arn_validator(cls, value):
         if not value.startswith("arn:aws:iam::"):
             raise ValueError(f"{value} is invalid role format. Please check your role again.")
         return value
+
+
+class DescriptionModel(BaseModel):
+    ProductTitle: str
+    ShortDescription: str
+    LongDescription: str
+    Sku: str
+    Highlights: List[str]
+    SearchKeywords: List[str]
+    Categories: List[str]
+
+
+class PromotionalResourcesModel(BaseModel):
+    LogoUrl: HttpUrl
+    Videos: List[HttpUrl]
+    AdditionalResources: List[
+        dict[
+            str,
+            str,
+        ]
+    ]
+
+    @field_validator("AdditionalResources")
+    def additional_resources_validator(cls, value) -> SupportResources:
+        return [{"Type": resource["Text"], "Url": HttpUrl(resource["Url"])} for resource in value]
+
+
+class SupportInformationModel(BaseModel):
+    Description: str
+    Resources: List[str]
+
+
+class RegionAvailabilityModel(BaseModel):
+    Regions: List[str]
+    FutureRegionSupport: str
+
+
+class EntityModel(BaseModel):
+    Description: DescriptionModel
+    PromotionalResources: PromotionalResourcesModel
+    SupportInformation: SupportInformationModel
+    RegionAvailability: RegionAvailabilityModel
+    SupportTerm: str
+
+    @staticmethod
+    def _get_entity(response: dict[str, Any]):
+        return EntityModel(**response)
+
+    @staticmethod
+    def _get_entity_from_yaml(yaml_config: dict[str, Any]):
+        desc = AmiProduct(**yaml_config["description"])
+        region = Region(**yaml_config["region"])
+        refund_policy = yaml_config["refund_policy"]
+
+        yaml_to_api_response = {
+            "Description": {
+                "ProductTitle": desc.product_title,
+                "ShortDescription": desc.short_description,
+                "LongDescription": desc.long_description,
+                "Sku": desc.sku,
+                "Highlights": desc.highlights,
+                "SearchKeywords": desc.search_keywords,
+                "Categories": desc.categories,
+            },
+            "PromotionalResources": {
+                "LogoUrl": desc.logourl,
+                "Videos": desc.video_urls,
+                "AdditionalResources": desc.additional_resources,
+            },
+            "SupportInformation": {
+                "Description": desc.support_description,
+                "Resources": desc.support_resources,
+            },
+            "RegionAvailability": {
+                "Regions": region.commercial_regions,
+                "FutureRegionSupport": region.future_region_support[-1],
+            },
+            "SupportTerm": refund_policy,
+        }
+
+        return EntityModel(**yaml_to_api_response)
+
+    def _get_dict(self, key: Optional[str]):
+        if key is None:
+            return self.dict()[key]
+
+        return self.dict()[key]
+
+    def _get_description_diff(self, local_entity: EntityModel):
```

Comment:
this function is a bit long. could we refactor for better readability? anywhere there is a comment describing the intended functionality is a good opportunity for functions